### DISCUSSION TITLE
Add input transform functions to improve the DX

### DISCUSSION THREAD

**Discussion by LcsGa** [primefaces/community#1044] [PrimeNG] [answered]
Since Angular v16.1, there is an [input `transform`](https://angular.dev/guide/components/inputs#input-transforms) function that you can implement to transform the incomming value from an input.
Angular already provides a some functions like `booleanAttribute` or `numberAttributes`.
Adding them would let us do:
```html
<p-knob showValue size="200" />
```
instead of:
```html
<p-knob [showValue]="true" [size]="200" />
```

To do so it would be as easy as doing this:
```ts
@Input({ transform: booleanAttribute }) showValue = false;

@Input({ transform: numberAttribute }) size = 100;
```

> NB: You could also even consider adding the [`required` input](https://angular.dev/guide/components/inputs#input-transforms) option to improve the DX even more, letting the devs know what is trully necessary when using components

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the feedback, we'll wait for community upvotes to implement them since they need to be added 90+ components.

**Reply by LcsGa** [reply]
Hi! I actually did it already: [here](https://github.com/primefaces/primeng/pull/14646#issuecomment-1913753823) and [here](https://github.com/primefaces/primeng/pull/14645#issuecomment-1913749367)

> NB: the first one is a branch made from the first one, I could probably remove the second one and rename the branch (since I forked it, I cannot make the frirst PR target the second one, and the second one target master
